src/bun.js/api/server.zig

```diff
@@ -6162,6 +6170,14 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
             request_value.ensureStillAlive();
 
             const response_value = this.config.onRequest.callWithThis(this.globalThis, this.thisObject, &args);
+            response_value.ensureStillAlive();
+            if (!response_value.isAnyError()) {
+                // JS signal is lazy created when needed and we need to keep it alive so we can call the events
+                const js_signal = RequestPrototype__signalGetCachedValue(request_value);
+                if (!js_signal.isEmptyOrUndefinedOrNull()) {
+                    ctx.js_signal = JSC.Strong.create(js_signal, this.globalThis);
```

Comment:
What do browsers do here? This feels strange to me

Reply:
`Request` owns signal (strong ref), in our case we ref the internal struct but not the JS counter part.
Another way of doing this is always strong ref inside Request when we create the signal, and in the server.zig side we ref count the request it self.